fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/BinaryImpl.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.models;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Collection;
+
+import org.fcrepo.kernel.api.Transaction;
+import org.fcrepo.kernel.api.exception.InvalidChecksumException;
+import org.fcrepo.kernel.api.models.Binary;
+import org.fcrepo.kernel.api.models.ResourceFactory;
+import org.fcrepo.kernel.api.services.policy.StoragePolicyDecisionPoint;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+
+
+/**
+ * Implementation of a Non-RDF resource.
+ *
+ * @author bbpennel
+ */
+public class BinaryImpl extends FedoraResourceImpl implements Binary {
+
+    /**
+     * Construct the binary
+     *
+     * @param id internal identifier
```

Comment:
We've used the term "fedora identifier" throughout to denote internal identifier.  For clarity, perhaps "fedora identifier" rather than "internal identifier"?